pgctl/cli.py

```diff
@@ -238,6 +239,10 @@ def __change_state(self, state):
                     for service in self.all_services
                 )
 
+        if self._entering_debug:
+            # Do not post-stop when we are about to start a debug session
+            run_post_stop_hook = False
+
```

Comment:
Then address that separately (making pre-start and post-stop consistent).  It's orthogonal to this review whose goal should be to call pre-start for the debug case.